CodeFileSanity/CodeSanityValidator.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace CodeFileSanity
+{
+    public class CodeSanityValidator
+    {
+
+        readonly string[] ignore_paths = {
+            "bin",
+            "obj",
+            "migrations",
+            "packages"
+        };
+
+        public bool HasErrors { get; private set; }
+
+        readonly bool isAppveyorBuild;
+
+        readonly string rootDirectory;
+
+        public CodeSanityValidator(ValidateCodeSanitySettings settings) 
+        {
+            isAppveyorBuild = settings.IsAppveyorBuild;
+            rootDirectory = settings.RootDirectory;
+        }
+
+        public void Validate() 
+            => checkDirectory(rootDirectory);
+
+        string getLicenseHeader(string path)
```

Comment:
missing `private` prefix

Reply:
As far I know the default access modificator is private in c#.
I mean I have nothing against changing.